From 5c70d91d515f4f1aa7ef4ebd2d3fa02a6ad09d3f Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Mon, 7 Aug 2017 14:58:09 +0800
Subject: [PATCH 1193/1345] fix: rtc: mvebu: skip error when set irq wake up
 state

commit  3fa3e7e2c003bff70387b7a01629a08ca6c31094 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

During RTC suspend and resume, the irq_set_irq_wake is called to
enable and disable the dedicated RTC wakeup state.
Finnally the irq_set_wake callback routine of parent irq chip will
be called for them.
However for GIC-400 irq chip used by RTC the irq_set_wake callback is
not implemented, which will return error.
In order that do not break the suspend and resume process, the return
value of routine enable_irq_wake() and disable_irq_wake() should
be ignored, which has same behaviour as most of RTC driver.

Change-Id: I76fe0693798dcdc07b15f8276b8c2c244d499ba9
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42621
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/rtc/rtc-armada38x.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/rtc/rtc-armada38x.c b/drivers/rtc/rtc-armada38x.c
index 21f355c..eae5723 100644
--- a/drivers/rtc/rtc-armada38x.c
+++ b/drivers/rtc/rtc-armada38x.c
@@ -466,7 +466,7 @@ static int armada38x_rtc_suspend(struct device *dev)
 	if (device_may_wakeup(dev)) {
 		struct armada38x_rtc *rtc = dev_get_drvdata(dev);
 
-		return enable_irq_wake(rtc->irq);
+		enable_irq_wake(rtc->irq);
 	}
 
 	return 0;
@@ -480,7 +480,7 @@ static int armada38x_rtc_resume(struct device *dev)
 		/* Update RTC-MBUS bridge timing parameters */
 		rtc->data->update_mbus_timing(rtc);
 
-		return disable_irq_wake(rtc->irq);
+		disable_irq_wake(rtc->irq);
 	}
 
 	return 0;
-- 
1.7.9.5

